src/Commands/core/UpdateDBCommands.php

```diff
@@ -323,30 +336,31 @@ public function cacheRebuild()
      *
      * @see \Drupal\system\Controller\DbUpdateController::batchFinished()
      * @see \Drupal\system\Controller\DbUpdateController::results()
+     *
+     * @param boolean $success Whether the batch ended without a fatal error.
+     * @param array $results
+     * @param array $operations
      */
     public function updateFinished($success, $results, $operations)
     {
-
         if (!$this->cache_clear) {
             $this->logger()->info(dt("Skipping cache-clear operation due to --no-cache-clear option."));
         } else {
             drupal_flush_all_caches();
         }
 
-        foreach ($results as $module => $updates) {
-            if ($module != '#abort') {
-                foreach ($updates as $number => $queries) {
-                    foreach ($queries as $query) {
-                        // If there is no message for this update, don't show anything.
-                        if (empty($query['query'])) {
-                            continue;
-                        }
+        // There's no success if any of the updates failed.
+        if (!empty($results['#abort'])) {
+            drush_set_option('updatedb_with_failures', TRUE);
```

Comment:
drush_set_option() is deprecated and brittle. Can we find a different way to do this? I'm happy to help once I understand the need.

Reply:
This is a way for the batch process to notify the caller. I've replaced it with a better approach. See the next commit.